angelleye-includes/express-checkout/class-wc-gateway-paypal-express-helper-angelleye.php

```diff
@@ -1056,7 +1056,7 @@ public function wc_ajax_update_shipping_costs() {
     public function angelleye_ajax_generate_cart() {
         global $wpdb, $post, $product;
         $product_id = '';
-        if (!wp_verify_nonce($_POST['nonce'], '_angelleye_generate_cart_nonce')) {
+        if ( isset($_POST['nonce']) && !wp_verify_nonce($_POST['nonce'], '_angelleye_generate_cart_nonce')) {
```

Comment:
Here since we are verifying the nonce so the condition should be written like this 
if ( !isset($_POST['nonce']) || !wp_verify_nonce($_POST['nonce'], '_angelleye_generate_cart_nonce'))

so that when nonce variable is not posted or when its posted then its validated. in current case if someone will remove nonce variable then this condition is optional and will not execute.